### PR TITLE
Use manifests cache in the tuist dump command

### PR DESCRIPTION
**Pull request by danibachar**
Resolves https://github.com/tuist/tuist/issues/6992

### Short description 📝

TL;DR see the [issue](https://github.com/tuist/tuist/issues/6992).
This is an alternative solution as suggested by @Ernest0-Production [here](https://github.com/tuist/tuist/pull/6993)

CC @Ernest0-Production and @fortmarek 

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Services/GenerateService.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ final class GenerateService {
         generatorFactory: GeneratorFactorying,
         clock: Clock = WallClock(),
         timeTakenLoggerFormatter: TimeTakenLoggerFormatting = TimeTakenLoggerFormatter(),
-        manifestLoader: ManifestLoading = ManifestLoader(),
+        manifestLoader: ManifestLoading = CachedManifestLoader(),
```
Comment (Ernest0-Production) [inline Sources/TuistKit/Services/GenerateService.swift:27]: I looked at the `GeneratorService` file in more detail and noticed that this variable is not used anywhere at all))

The manifest loader is taken from `generatorFactory`. And there it returns `CachedManifestLoader` by default.

At the same time, the change in `DumpService` is advisable.

**Review by fortmarek — APPROVED** [review]
